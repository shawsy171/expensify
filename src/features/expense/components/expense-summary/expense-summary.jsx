import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';

// selectors
import selectorExpenses, { selectorExpensesTotal } from './../../store/expense/selectors';

export const ExpenseSummary = ({expenseTotal, expenseCount}) => {

    const formatTotal = (num) => ('Â£' + numeral(num / 100).format('0,0.00'))
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';

    return (
        <div>
            <h2>Viewing {expenseCount} {expenseWord} totalling {formatTotal(expenseTotal)}</h2>
        </div>
    );
}
const mapStateToProps = state => {
    const visibleExpences = selectorExpenses(state.expenses, state.filters);
    return {
        expenseCount: visibleExpences.length,
        expenseTotal: selectorExpensesTotal(visibleExpences),
    }
}
export default connect(mapStateToProps)(ExpenseSummary);