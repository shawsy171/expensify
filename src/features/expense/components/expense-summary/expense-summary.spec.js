import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from './expense-summary';

describe('Expense Summary', () => {

    test('should show 1 expense message', () => {
        const wrapper = shallow(<ExpenseSummary expenseTotal={12345} expenseCount={1} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should show multiple expenses message', () => {
        const wrapper = shallow(<ExpenseSummary expenseTotal={12345} expenseCount={4} />);
        expect(wrapper).toMatchSnapshot();
    });

})
