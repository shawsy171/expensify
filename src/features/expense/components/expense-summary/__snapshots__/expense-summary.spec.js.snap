// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expense Summary should show 1 expense message 1`] = `
<div>
  <h2>
    Viewing 
    1
     
    expense
     totalling 
    £123.45
  </h2>
</div>
`;

exports[`Expense Summary should show multiple expenses message 1`] = `
<div>
  <h2>
    Viewing 
    4
     
    expenses
     totalling 
    £123.45
  </h2>
</div>
`;
